source/distributed/field_transfer.cc

```diff
@@ -34,12 +34,96 @@ namespace parallel
         const DoFHandler<dim, spacedim> &dof)
         : dof_handler(dof)
       {
+        // When coarsening, we want to mimic the behavior of SolutionTransfer
+        // and interpolate from child cells to parent. Define this strategy here
+        // since it is not readily available
+        const auto coarsening_strategy =
+          [this](
+            const typename dealii::Triangulation<dim, spacedim>::cell_iterator
+              &                                parent,
+            const std::vector<Vector<Number>> &children_values) {
+            // get the equivalent DoFCellAccessor
+            typename DoFHandler<dim, spacedim>::cell_iterator dof_cell_iterator(
+              &dof_handler.get_triangulation(),
+              parent->level(),
+              parent->index(),
+              &dof_handler);
+
+            int fe_index = 0;
+            if (dof_handler.has_hp_capabilities())
+              fe_index = dealii::internal::hp::DoFHandlerImplementation::
+                dominated_future_fe_on_children<dim, spacedim>(
+                  dof_cell_iterator);
+
+            const auto &fe = dof_handler.get_fe(fe_index);
+            Assert(fe.n_dofs_per_cell() > 0,
+                   ExcMessage(
+                     "Cannot coarsen onto a FiniteElement with no DoFs."));
```

Comment:
Is that true even if all the children cells use `FE_Nothing`?

Reply:
I think the FE here (returned by the domination logic) should never be FE_Nothing.